src/supernova/index.ts

```diff
@@ -0,0 +1,345 @@
+  (v, i) => v * xsectionESeAnu[i] * neutrinoTargets
+);
+export const sumSpectrumAnuESEforIO =
+  sum(eventSpectrumAnuESEforIO) * deltaEnergy;
+
+// x2 for mu and tau
+export const xsectionESeNux = energyValues.map(function (x) {
+  return crossSectionElasticScattering(
+    x,
+    NeutrinoType.muTauNeutrino,
+    undefined,
+    undefined,
+    NeutrinoTarget.electron
+  );
+});
+export const eventSpectrumNuxESEforNO = oscillatedFluxSpectrums[NeutrinoType.muTauNeutrino][MassOrdering.Normal].map(
+  (v, i) => v * xsectionESeNux[i] * neutrinoTargets
+);
+export const sumSpectrumNuxESEforNO =
+  sum(eventSpectrumNuxESEforNO) * deltaEnergy * 2;
+export const eventSpectrumNuxESEforIO = oscillatedFluxSpectrums[NeutrinoType.muTauNeutrino][MassOrdering.Inverted].map(
+  (v, i) => v * xsectionESeNux[i] * neutrinoTargets
+);
+export const sumSpectrumNuxESEforIO =
+  sum(eventSpectrumNuxESEforIO) * deltaEnergy * 2;
+
+// x2 for mu and tau
+export const xsectionESeAnx = energyValues.map(function (x) {
+  return crossSectionElasticScattering(
+    x,
+    NeutrinoType.muTauAntineutrino,
+    undefined,
+    undefined,
+    NeutrinoTarget.electron
+  );
+});
+export const eventSpectrumAnxESEforNO = oscillatedFluxSpectrums[NeutrinoType.muTauNeutrino][MassOrdering.Normal].map(
+  (v, i) => v * xsectionESeAnx[i] * neutrinoTargets
+);
+export const sumSpectrumAnxESEforNO =
+  sum(eventSpectrumAnxESEforNO) * deltaEnergy * 2;
+export const eventSpectrumAnxESEforIO = oscillatedFluxSpectrums[NeutrinoType.muTauNeutrino][MassOrdering.Inverted].map(
+  (v, i) => v * xsectionESeAnx[i] * neutrinoTargets
+);
+export const sumSpectrumAnxESEforIO =
+  sum(eventSpectrumAnxESEforIO) * deltaEnergy * 2;
+
+
+//-----
+
+interface CEvNSEventsInterface {
+  [NeutrinoType.electronNeutrino]: number;
+  [NeutrinoType.electronAntineutrino]: number;
+  [NeutrinoType.muTauNeutrino]: number; // both... Anti/Anti-anti
+}
+
+export const CEvNSEvents = (element: Element, TMin:number): CEvNSEventsInterface => {
+  let targetParams = {tMin: TMin, ...getTargetParamsCEvNS(element)};
+  let xsectionCEvNS = energyValues.map((ev) => xSectionCEvNS(ev, targetParams));
+  let eventSpectrumNueCEvNS = fluxSpectrums[NeutrinoType.electronNeutrino].map(
+    (v, i) => v * xsectionCEvNS[i] * targetParams.nuclearTargets
+  );
+  let eventSpectrumAnuCEvNS = fluxSpectrums[NeutrinoType.electronAntineutrino].map(
+    (v, i) => v * xsectionCEvNS[i] * targetParams.nuclearTargets
+  );
+  let eventSpectrumNuxCEvNS = fluxSpectrums[NeutrinoType.muTauNeutrino].map(
+    (v, i) => v * xsectionCEvNS[i] * targetParams.nuclearTargets
+  );
+  return {
+    [NeutrinoType.electronNeutrino]: sum(eventSpectrumNueCEvNS) * deltaEnergy,
+    [NeutrinoType.electronAntineutrino]:
+      sum(eventSpectrumAnuCEvNS) * deltaEnergy,
+    [NeutrinoType.muTauNeutrino]: sum(eventSpectrumNuxCEvNS) * deltaEnergy * 4,
+  };
+};
+
+export const CEvNSxSectionAr40 = energyValues.map(function (x) {
```

Comment:
What is this for? If for plotting, it should be in the plot component directly. Unless you want the tMin param to be user modifiable.

Reply:
It is for the bottom plot showing the various cross sections.